src/OrchardCore.Modules/OrchardCore.Media.Indexing.Pdf/Services/PdfMediaFileTextProvider.cs

```diff
@@ -11,16 +11,15 @@ public async Task<string> GetTextAsync(string path, Stream fileStream)
         // https://github.com/UglyToad/PdfPig/blob/master/src/UglyToad.PdfPig.Core/StreamInputBytes.cs#L45.
         // Thus if it isn't, which is the case with e.g. Azure Blob Storage, we need to copy it to a new, seekable
         // Stream.
-        MemoryStream seekableStream = null;
+        FileStream seekableStream = null;
         try
         {
             if (!fileStream.CanSeek)
             {
-                // Since fileStream.Length might not be supported either, we can't preconfigure the capacity of the
-                // MemoryStream.
-                seekableStream = new MemoryStream();
-                // While this involves loading the file into memory, we don't really have a choice.
+                seekableStream = new FileStream(Path.GetTempFileName(), FileMode.Create, FileAccess.Write, FileShare.None, 4096, FileOptions.DeleteOnClose | FileOptions.Asynchronous);
```

Comment:
@hishamco please check the differences between `GetTempFileName()` and `GetRandomFileName()`, you'll see what was wrong with your combine.

Reply:
I didn't use `GetRandomFileName()` :)